packages/web/src/pages/upload-page/forms/UploadTrackForm.tsx

```diff
@@ -30,19 +35,23 @@ export const UploadTrackForm = (props: UploadTrackFormProps) => {
       tracks: tracks as TrackForUpload[],
       trackMetadatas: tracks.map((track) => ({
         ...track.metadata,
-        description: '',
-        releaseDate: new Date(moment().toString()),
-        tags: '',
+        ...initialMetadata,
+        description: initialMetadata?.description ?? '',
+        releaseDate: initialMetadata?.release_date
+          ? new Date(initialMetadata.release_date)
+          : new Date(moment().toString()),
+        tags: initialMetadata?.tags ?? '',
         field_visibility: {
           ...defaultHiddenFields,
+          ...initialMetadata?.field_visibility,
           remixes: true
         },
-        stems: [],
-        isrc: '',
-        iswc: ''
+        stems: initialMetadata?.stems ?? [],
```

Comment:
all a bit funky but i get it

Reply:
open to ideas, maybe we move `stems` above the `initialMetadata` spread?